apps/condo/package.json

```diff
@@ -166,6 +166,7 @@
     "react-yandex-metrika": "^2.6.0",
     "recheck": "^4.4.5",
     "redlock": "^5.0.0-beta.2",
+    "rubles": "^0.2.0",
```

Comment:
This is a very old unsupported library, maybe there is something newer? 
If not, a suitable one, then you can leave only numbers in the document, without converting to words

And is there a package with support for different languages?

Reply:
There is a package with support for different currencies, newer and with normal generation of kopecks. https://github.com/vicimpa/vicimpa-library

Reply:
Why not simply inline the code?

Reply:
The library is very cool. There are many useful functions. In general, yes, it can be rewritten. If you don't like the library, I'll rewrite it.

Reply:
> Why not simply inline the code? It is one script, like 100 characters ...

No need to write your own library...
I doubt it will fit in 100 characters

Reply:
Let's leave only numbers then, without transforming them into words. This option will also be fine

Reply:
I'm also in only numbers club, but it is expected in Ru so that you'd have this `words` near numbers :-( 

--

**How to approach this?**

I'm a bit against on introducing a new library just for the task, 

when introducing a new library you add one more supply chain attack source and bloat your node_modules (so you now have to download more stuff from npm every time you run `npm i`)

Solution i'd like will be to simply inline the library inside the document generator code, and maybe add some tests. This way you'd incapsulate this logic in this feature and wouldn't introduce a new dependency

-- 

However this solution is bad if:

- You need to do this `number-to-text` transform anywhere else, for example in billing code
- You need this code to work in non russian environment

In this scenario its wise to add one more dependency. 

- Dependency should support major locales
- Code should be OK (have tests, etc)
- Dependency should be maintained well – check issues and pull request. Does author reply to them?
- Dependency should be small. No need to drag 5mb of JS code into our codebase
- The less dependencies this dependency have – the better
- License should be permissive. Best one out there – MIT

The dependency named [ruble](https://www.npmjs.com/package/@vicimpa/rubles) will not be ok for use, becuase it doesnt support 

The package I kinda like is this: https://www.npmjs.com/package/to-words but it doesnt support RU-ru locale, so you might want to do a PR there first :-)

--

In a nutshell
- Consult legal team. Can you ship the feature without number-to-words?
- Consult billing team. Do they have the need for a separate package that does number-to-words?
- If billing team does not need the solution, i'd have inlined the code to the generator (it is RU-ru anyway) and would've added some tests, so that it is easier for us to support it
- If billing team needs the solution – you'd need to choose package and maybe create a package in open condo

Reply:
> but it is expected in Ru so that you'd have this `words` near numbers :-(

we were told a long time ago that it is not necessary to translate numbers into words :)
In my understanding, this moment only delays the release of the feature. If you really want, you can do it later. What do you think about this? :)

If you want to do it, do it well, so that your implementation is well tested and can support implementation for different locales (right now you can do only ru, but adding other languages should not be a problem).
And it is definitely not "100 characters", as you said earlier)))

And also if you copy code from another repository, then it would probably be nice to leave a link to the source :)

Reply:
> The package I kinda like is this: https://www.npmjs.com/package/to-words but it doesnt support RU-ru locale, so you might want to do a PR there first :-)

Yeah, I saw it too and I liked the library, but there is no ru locale. And I looked at the pull request list - there are two requests for adding locales that are over a year old, so I don't think adding ru locale will be a quick thing :(